File: server/src/models/user_model.js
const { Schema, model } = require("mongoose");

const userSchema = new Schema({
  fullName: { type: String, required: true, default: "" },
  email: { type: String, unique: true, required: true, default: "" },
  phoneNumber: { type: String, unique: true, default: "" },
  address: { type: String, default: "" },
  createdOn: { type: Date },
  updatedOn: { type: Date },
});

userSchema.pre("save", function (next) {
  this.updatedOn = new Date();
  this.createdOn = new Date();
  next();
});

const UserModel = model("User", userSchema);

module.exports = UserModel;

File: lib/data/models/user_model.dart
class UserModel {
  String? fullName;
  String? email;
  String? phoneNumber;
  String? address;
  String? sId;
  String? updatedOn;
  String? createdOn;

  UserModel(
      {this.fullName,
      this.email,
      this.phoneNumber,
      this.address,
      this.sId,
      this.updatedOn,
      this.createdOn});

  UserModel.fromJson(Map<String, dynamic> json) {
    fullName = json['fullName'];
    email = json['email'];
    phoneNumber = json['phoneNumber'];
    address = json['address'];
    sId = json['_id'];
    updatedOn = json['updatedOn'];
    createdOn = json['createdOn'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['fullName'] = this.fullName;
    data['email'] = this.email;
    data['phoneNumber'] = this.phoneNumber;
    data['address'] = this.address;
    data['_id'] = this.sId;
    data['updatedOn'] = this.updatedOn;
    data['createdOn'] = this.createdOn;
    return data;
  }
}
